File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0;
    let minPrice = prices[0];
    prices.forEach(price => {
        profit = price - minPrice;
        minPrice = Math.min(price, minPrice);
        maxProfit = Math.max(profit, maxProfit);
    })
    return maxProfit;
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.rb
# @param {Integer[]} prices
# @return {Integer}
def max_profit(prices)
    max_profit = 0
    min_price = prices[0]
    prices.each do |price| 
        profit = price - min_price
        max_profit = [max_profit, profit].max
        min_price = [min_price, price].min
    end
    max_profit
end